src/main/java/com/github/shyiko/mysql/binlog/BinaryLogClient.java

```diff
@@ -1034,13 +1024,22 @@ private void updateGtidSet(Event event) {
                 GtidEventData gtidEventData = (GtidEventData) EventDataWrapper.internal(event.getData());
                 gtid = gtidEventData.getGtid();
                 break;
+            case MARIADB_GTID:
+                MariadbGtidEventData mariadbGtidEventData =  (MariadbGtidEventData) EventDataWrapper.internal(event.getData());
+                mariadbGtidEventData.setServerId(eventHeader.getServerId());
+                gtid = mariadbGtidEventData.toString();
+                break;
+            case MARIADB_GTID_LIST:
+                MariadbGtidListEventData mariadbGtidListEventData =  (MariadbGtidListEventData) EventDataWrapper.internal(event.getData());
+                gtid = mariadbGtidListEventData.getMariaGTIDSet().toString();
+                break;
             case XID:
                 commitGtid();
                 tx = false;
                 break;
             case QUERY:
-                QueryEventData queryEventData = (QueryEventData) EventDataWrapper.internal(event.getData());
-                String sql = queryEventData.getSql();
+            case ANNOTATE_ROWS:
+                String sql = getQuerySql(event);
```

Comment:
if you look at the implementation of `getQuerySql`, you'll see that this is false sharing -- in the case of MariaDB's `QUERY` event it bounces right back to the mysql-compatible `getQuerySql`.

Therefore -- keep the implementation of the `QUERY` case the same, just make a different impl for maria.